django/contrib/admin/options.py

```diff
@@ -2138,17 +2138,20 @@ def has_delete_permission(self, request, obj=None):
         return super().has_delete_permission(request, obj)
 
     def has_view_permission(self, request, obj=None):
-        if self.opts.auto_created:
-            opts = self.opts
+        opts = self.opts
+        if opts.auto_created:
```

Comment:
Again, this probably counts as unrelated.

(Not sure it's cleaner either: `opts` is now in scope outside the `if` block...)